### PR TITLE
Tcs 20 add segmented subscriptions filter and only calculate symbol segments for new symbols when mapping

### PR DESCRIPTION
**Pull request by AdamBrown2**
Added the ignoredtables.csv for system tables to be ignored and always subscribed to.
Adapted .stpps.subsegment to use the ignore list

Bug Fix
error message was changed for .stpps.subsegment so that .rdb.subtables is populated correctly as symbol list and not mixed.
Added if statement to rdb.q for datastriping to redefine the ignorelist to include tables that have not been subscribed to. This stops the .rdb.reload from reloading the tables that are not subscribed to.

### CODE REVIEW

**Comment by AdamBrown2** [comment]
Commit d5e108b has all the changes

File: code/common/pubsub.q [hunk 1]
```diff
@@ -160,36 +160,22 @@ init:{[t]
...
-                `.ds.subreq set .ds.subreq,' .ds.map[.ds.numseg;sym where new];
-            ];
-        ];
-        // Initialize .ds.subreq if does not exists
-        `.ds.subreq set .ds.map[.ds.numseg;sym]
-    ];
-    .ds.subreq skey
-    }
+  // If no updates, return
+  if[0=count input;:`boolean$()];
```
Comment (drgdavies) [inline code/common/pubsub.q:172]: What is the purpose of returning null boolean list?

File: code/common/subscriptions.q [hunk 2]
```diff
@@ -62,7 +62,7 @@ replay0:{[tabs;realsubs;schemalist;logfilelist;filters]
   origupd:@[value;`..upd;{{[x;y]}}];
   // only use tables user has access to
   subtabs:realsubs[`subtabs];
-  if[count where nullschema:0=count each schemalist;
+  if[count where nullschema:0=count each schemalist; // change needed here
```
Comment (drgdavies) [inline code/common/subscriptions.q:65]: what is the change needed here and when is it being done, is it post master push or now?

File: code/processes/rdb.q [hunk 3]
```diff
@@ -112,7 +112,7 @@ endofday:{[date;processdata]
 	hdbs:distinct raze {exec w from .servers.getservers[x;y;()!();1b;0b]}'[`proctype`procname;(hdbtypes;hdbnames)];
 	notifyhdb[;date] each hdbs;
 	};
-	
+
```
Comment (drgdavies) [inline code/processes/rdb.q:115]: Remove extra line

File: code/processes/wdb.q [hunk 4]
```diff
@@ -372,13 +377,24 @@ endofdaysort:{[dir;pt;tablist;writedownmode;mergelimits;hdbsettings]
...
-		{[x;y;ptype].[{neg[y]@x};(x;y);{[ptype;x].lg.e[`reloadproc;"failed to reload the ",string[ptype]];'x}[ptype]]}[({@[`. `reload;x;()]; (neg .z.w)(`.wdb.handler;1b); (neg .z.w)[]};d);h;ptype];
-		@[h;(`reload;d);{[ptype;e] .lg.e[`reloadproc;"failed to reload the ",string[ptype],".  The error was : ",e]}[ptype]]
-	];
-	.lg.o[`reload;"the ",string[ptype]," has been successfully reloaded"];
-	}
+        /-count of processes to be reloaded 
+        .wdb.countreload:count[raze .servers.getservers[`proctype;;()!();1b;0b]each reloadorder];
+        /-defining lambdas to be in asynchronously calling processes to reload 
+        /-async call back function executed when eodwaittime>0
+        sendfunc:{[x;y;ptype].[{neg[y]@x};(x;y);{[ptype;x].lg.e[`reloadproc;"failed to reload the ",string[ptype]];'x}[ptype]]};
```
Comment (drgdavies) [inline code/processes/wdb.q:384]: Any reason why we are doing these as lambdas in side the function?

**Pull request closed without merging**